src/setup-uv.ts

```diff
@@ -69,38 +70,30 @@ async function setupUv(
   checkSum: string | undefined,
   githubToken: string,
 ): Promise<{ uvDir: string; version: string }> {
-  let installedPath: string | undefined;
-  let cachedToolDir: string;
-  let version: string;
-  if (versionInput === "latest") {
-    const latestResult = await downloadLatest(
-      platform,
-      arch,
-      checkSum,
-      githubToken,
-    );
-    version = latestResult.version;
-    cachedToolDir = latestResult.cachedToolDir;
-  } else {
-    const toolCacheResult = tryGetFromToolCache(arch, versionInput);
-    version = toolCacheResult.version;
-    installedPath = toolCacheResult.installedPath;
-    if (installedPath) {
-      core.info(`Found uv in tool-cache for ${versionInput}`);
-      return { uvDir: installedPath, version };
-    }
-    const versionResult = await downloadVersion(
-      platform,
-      arch,
-      versionInput,
-      checkSum,
-      githubToken,
-    );
-    cachedToolDir = versionResult.cachedToolDir;
-    version = versionResult.version;
+  const resolvedVersion = await resolveVersion(
+    versionInput === "latest"
+      ? await getLatestReleaseVersion(githubToken)
+      : versionInput,
+    githubToken,
+  );
+
+  const toolCacheResult = tryGetFromToolCache(arch, resolvedVersion);
```

Comment:
I've also fixed another unexpected behavior.

- Previously, the version argument passed to `tryGetFromToolCache` could be in the X.Y format, unlike the `X.Y.Z` format used by the `downloadVersion` function (referred to as `resolvedVersion`). 
- This meant that `tryGetFromToolCache` cannot find a cached version when given a `X.Y` version string, as the download functionality only supports resolvedVersion.
  - https://github.com/astral-sh/setup-uv/blob/8bdd012be50ea391c968b486f6510cfe82922108/src/download/download-version.ts#L64-L69

Therefore, I've modified the code to pass `resolvedVersion` to `tryGetFromToolCache`.

Reply:
Given we already have an explicit version at this point because we now call `resolveVersion` before we can remove some stuff inside `tryGetFromToolCache`. It should then only contain:

```typescript
const installedPath = tc.find(TOOL_CACHE_NAME, resolvedVersion, arch);
return { version: resolvedVersion, installedPath };
```
---
Also:

Have you verified your the unexpected behavior you described? According to the docs it should work with `X.Y`: https://github.com/actions/toolkit/tree/main/packages/tool-cache#find

Reply:
@eifinger Oh! I misunderstood the essence of the problem.
You are correct about the `tryGetFromToolCache` behavior. I missed tc.evaluateVersions. Sorry about that.

---

I have another question regarding the previous code. What is the expected behavior in this situation?

1. Run setup-uv with version: 0.5
   - 0.5.55 (latest at this time) is installed.
2. A new release of UV is available (0.5.56).
3. Run setup-uv with version: 0.5 (before the cache is cleared).
   - **Will the cached version (0.5.55) be used, or will 0.5.56 (the latest version) be installed?**

Reply:
This can be confirmed by doing this workflow.

```yaml
    steps:
      - uses: actions/checkout@v4
      - name: Install default version
        uses: ./
        with:
          version: "0.5.4"
      - name: Use cache or install latest 0.5.5
        uses: ./
        with:
          version: "0.5"
      - name: Check that uv is installed
        run: |
          uv --version
```

<details>

<summary>previous code uses the cached one (0.5.4)</summary>

checked in https://github.com/pollenjp/setup-uv/actions/runs/12092953102/job/33723332583?pr=3

![image](https://github.com/user-attachments/assets/e89d3f0d-76da-4cc5-ac7e-2478e95371d1)

</details>


<details>

<summary>my code install the latest one (0.5.5)</summary>

checked in https://github.com/pollenjp/setup-uv/actions/runs/12092943568/job/33723311457

![image](https://github.com/user-attachments/assets/0cc20e68-b120-42b5-9b7d-5b739b8bcf2b)

</details>